tartiflette/language/validators/query/field_selections_on_objects_interfaces_and_unions_types.py

```diff
@@ -0,0 +1,31 @@
+from tartiflette.utils.errors import graphql_error_from_nodes
+
+
+class FieldSelectionsOnObjectsInterfacesAndUnionsTypes:
+    RULE_NAME = "field-selections-on-objects-interfaces-and-unions-types"
+
+    def __init__(self):
+        self._extensions = {
+            "rule": "5.3.1",
+            "spec": "June 2018",
+            "details": "https://graphql.github.io/graphql-spec/June2018/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types",
+            "tag": self.RULE_NAME,
+        }
+
+    def validate(self, target, path, **__):
+        graphql_type, parent_type_name, field = target
+
+        if field.name.value.startswith("__"):
+            return []
+
+        if graphql_type is None:
+            return [
+                graphql_error_from_nodes(
+                    message=f"Field {field.name.value} doesn't exist on {parent_type_name or 'Root'}",
```

Comment:
Are we agree that `Root` can only be returned if we already are on a non-existing parent field?

Reply:
Yes. This is ugly but I think this doesn't happen anymore, I'll have to check but I think parent type is always set now.